modules/auxiliary/gather/mikrotik_winbox_fileread.py

```diff
@@ -0,0 +1,163 @@
+#!/usr/bin/env python3
+# -*- coding: utf-8 -*-
+
+# standard modules
+import logging
+
+# extra modules
+dependencies_missing = False
+try:
+    import socket
+    import sys
+    import hashlib
+except ImportError:
+    dependencies_missing = True
+
+from metasploit import module
+
+metadata = {
+    'name': 'Mikrotik Winbox Arbitrary File Read',
+    'description': '''
+        MikroTik RouterOS (bugfix) 6.30.1-6.40.7, (current) 6.29-6.42, (RC) 6.29rc1-6.43rc3 allows unauthenticated
+        remote attackers to read arbitrary files through a directory traversal through the WinBox interface
+        (typically port 8291).''',
+    'authors': [
+        'mosajjal', # PoC
+        'h00die' # msf port
+    ],
+    'date': '2018-08-02',
+    'license': 'MSF_LICENSE',
+    'references': [
+        {'type': 'url', 'ref': 'https://github.com/BasuCert/WinboxPoC'},
+        {'type': 'url', 'ref': 'https://blog.n0p.me/2018/05/2018-05-21-winbox-bug-dissection/'},
+        {'type': 'url', 'ref': 'https://blog.mikrotik.com/security/winbox-vulnerability.html'},
+        {'type': 'cve', 'ref': '2018-14847'},
+        {'type': 'edb', 'ref': '45578'}
+    ],
+    'type': 'single_scanner',
+    'options': {
+        'rhost': {'type': 'address', 'description': 'Target address', 'required': True, 'default': None},
+        'rport': {'type': 'port', 'description': 'Target port', 'required': True, 'default': 8291},
```

Comment:
^ cc @acammack-r7 - are there any requirements to use a specific capitalisation of RHOST / rhost for external modules?

A quick scan of the codebase shows that we've got a mix of thing of approaches:
https://github.com/rapid7/metasploit-framework/blob/5fbe243662390054fc4dcc4f8250b85068709152/modules/auxiliary/admin/http/grafana_auth_bypass.py#L46-L49
https://github.com/rapid7/metasploit-framework/blob/5fbe243662390054fc4dcc4f8250b85068709152/modules/exploits/windows/smb/ms17_010_eternalblue_win8.py#L86-L89
https://github.com/rapid7/metasploit-framework/blob/5fbe243662390054fc4dcc4f8250b85068709152/modules/auxiliary/admin/http/cisco_7937g_ssh_privesc.py#L39-L41

Whilst the docs show lowercase usage:
https://github.com/rapid7/metasploit-framework/wiki/Writing-External-Python-Modules#full-example

Which approach should be preferenced? 🔍 

I haven't verified this, but I seem to remember that maybe the python datastore is case sensitive unlike the Ruby datastore, so this inconsistency might be confusing to users? I'll create a separate ticket for insensitive dict support that if that is the case though 😄

Reply:
For context, it was this picture of the module options from @cgranleese-r7 that caught my eye:

![image](https://user-images.githubusercontent.com/60357436/97993294-c7ad5180-1ddb-11eb-99c1-c08d8d43f670.png)

Reply:
I sourced the example in the docs, so any decision here should be replicated there as well.

Reply:
Looks like for external modules we'd want to preference not using screaming case. So this usage, and the example code from the wiki is perfect. What we're maybe missing is the Ruby layer converting the case insensitive datastore into the same casing as what the external module requires it to be, which is definitely an unrelated issue to this pull request 😄